tests/unit/amo/reducers/test_addonsByAuthors.js

```diff
@@ -38,120 +38,173 @@ describe(__filename, () => {
-      }).toThrow(/An errorHandlerId is required/);
-    });
+    it('adds each add-on to each author array', () => {
+      const multiAuthorAddon = {
+        ...fakeAddon,
+        id: 501,
+        authors: [
+          { ...fakeAuthor, id: 50 },
+          { ...fakeAuthor, id: 60 },
+        ],
+      };
+      const params = getParams({
+        addons: [multiAuthorAddon],
+      });
 
-    it('requires a slug', () => {
-      const params = getParams();
-      delete params.slug;
-      expect(() => {
-        fetchOtherAddonsByAuthors(params);
-      }).toThrow(/An add-on slug is required/);
+      const newState = reducer(undefined, loadAddonsByAuthors(params));
+
+      expect(newState.byUserId).toEqual({
+        50: [multiAuthorAddon.id],
+        60: [multiAuthorAddon.id],
+      });
     });
 
-    it('requires an add-on type', () => {
-      const params = getParams();
-      delete params.addonType;
-      expect(() => {
-        fetchOtherAddonsByAuthors(params);
-      }).toThrow(/An add-on type is required/);
+    it('adds each different add-on to the byAddonId dictionary', () => {
+      const firstAddon = {
+        ...fakeAddon,
+        id: 6,
+        authors: [
+          { username: 'test', id: 51 },
+          { username: 'test2', id: 61 },
+        ],
+      };
+      const secondAddon = {
+        ...fakeAddon, id: 7, authors: [{ username: 'test2', id: 61 }],
+      };
+      const thirdAddon = {
+        ...fakeAddon, id: 8, authors: [{ username: 'test3', id: 71 }],
+      };
+      const params = getParams({
+        addons: [firstAddon, secondAddon, thirdAddon],
+        forAddonSlug: undefined,
+      });
+
+      const newState = reducer(undefined, loadAddonsByAuthors(params));
+
+      expect(newState.byAddonId).toEqual({
+        [firstAddon.id]: createInternalAddon(firstAddon),
+        [secondAddon.id]: createInternalAddon(secondAddon),
+        [thirdAddon.id]: createInternalAddon(thirdAddon),
+      });
     });
 
-    it('requires some authors', () => {
-      const params = getParams();
-      delete params.authors;
-      expect(() => {
-        fetchOtherAddonsByAuthors(params);
-      }).toThrow(/Authors are required/);
+    it('adds each different add-on to each author array', () => {
+      const firstAddon = {
+        ...fakeAddon,
+        id: 6,
+        authors: [
+          { username: 'test', id: 51 },
+          { username: 'test2', id: 61 },
+        ],
+      };
+      const secondAddon = {
+        ...fakeAddon, id: 7, authors: [{ username: 'test2', id: 61 }],
+      };
+      const thirdAddon = {
+        ...fakeAddon, id: 8, authors: [{ username: 'test3', id: 71 }],
+      };
+      const params = getParams({
+        addons: [firstAddon, secondAddon, thirdAddon],
+        forAddonSlug: undefined,
+      });
+
+      const newState = reducer(undefined, loadAddonsByAuthors(params));
+
+      expect(newState.byUserId).toEqual({
+        51: [firstAddon.id],
```

Comment:
It would be helpful to see `51`, `61`, and `71` replaced by short variable names.